src/diffusers/pipelines/marigold/pipeline_marigold_depth.py

```diff
@@ -0,0 +1,661 @@
+        resample_method_output: str,
+        batch_size: int,
+        ensembling_kwargs: Optional[Dict[str, Any]],
+        latents: Optional[torch.FloatTensor],
+        generator: Optional[Union[torch.Generator, List[torch.Generator]]],
+        output_prediction_format: str,
+        output_visualization_kwargs: Optional[Dict[str, Any]],
+    ) -> None:
+        if num_inference_steps is None:
+            raise ValueError("`num_inference_steps` is not specified and could not be resolved from the model config.")
+        if num_inference_steps < 1:
+            raise ValueError("`num_inference_steps` must be positive.")
+        if ensemble_size < 1:
+            raise ValueError("`ensemble_size` must be positive.")
+        if ensemble_size == 2:
+            logger.warning(
+                "`ensemble_size` == 2 results are similar to no ensembling (1); "
+                "consider increasing the value to at least 3."
+            )
+        if processing_resolution is None:
+            raise ValueError(
+                "`processing_resolution` is not specified and could not be resolved from the model config."
+            )
+        if processing_resolution < 0:
+            raise ValueError(
+                "`processing_resolution` must be non-negative: 0 for native resolution, or any positive value for "
+                "downsampled processing."
+            )
+        if processing_resolution % self.vae_scale_factor != 0:
+            raise ValueError(f"`processing_resolution` must be a multiple of {self.vae_scale_factor}.")
+        if resample_method_input not in ("nearest", "nearest-exact", "bilinear", "bicubic", "area"):
+            raise ValueError(
+                "`resample_method_input` takes string values compatible with PIL library: "
+                "nearest, nearest-exact, bilinear, bicubic, area."
+            )
+        if resample_method_output not in ("nearest", "nearest-exact", "bilinear", "bicubic", "area"):
+            raise ValueError(
+                "`resample_method_output` takes string values compatible with PIL library: "
+                "nearest, nearest-exact, bilinear, bicubic, area."
+            )
+        if batch_size < 1:
+            raise ValueError("`batch_size` must be positive.")
+        if output_prediction_format not in ["pt", "np", "pil"]:
+            raise ValueError("`output_prediction_format` must be one of `pt`, `np`, or `pil`.")
+        if latents is not None and generator is not None:
+            raise ValueError("`latents` and `generator` cannot be used together.")
+        if ensembling_kwargs is not None and not isinstance(ensembling_kwargs, dict):
+            raise ValueError("`ensembling_kwargs` must be a dictionary.")
+        if output_visualization_kwargs is not None and not isinstance(output_visualization_kwargs, dict):
+            raise ValueError("`output_visualization_kwargs` must be a dictionary.")
+
+        # image checks
+        num_images = 1
```

Comment:
I think it will still be much nicer to allow multiple images and throw an error if size does not match; 
it will be nice to make sure this pipeline works with all the output format of our other pipelines, this way users can chain depth pipeline with our other pipelines and use the outputs from other pipeline as inputs here 

our pipelines image outputs include a list of pil image so it would be nice to make it work here

Reply:
Yes, I changed the image preprocessor and pipeline input type to accept the standard `PipelineImageInput`, which can be PIL,np,torch, or lists of them. If a list is detected, the preprocessor raises exception when the images cannot be stacked